orchestrator/test_runner/src/airdrop_proposal.rs

```diff
@@ -219,3 +215,43 @@ async fn wait_for_proposals_to_execute(contact: &Contact) {
         sleep(Duration::from_secs(5)).await;
     }
 }
+
+fn generate_accounts_and_amounts(rng: &mut ThreadRng) -> (Vec<CosmosAddress>, Vec<u64>) {
+    // Generate user keys for the airdrop, converting between private key and address
+    // is quite slow, so we skip that step and go directly to an address
+    let mut user_addresses = Vec::new();
+    let mut amounts: Vec<u64> = Vec::new();
+    for _ in 0..NUM_AIRDROP_RECIPIENTS {
+        let secret: [u8; 20] = rng.gen();
+        let amount: u32 = rng.gen();
+        let cosmos_address = CosmosAddress::from_bytes(secret, ADDRESS_PREFIX.as_str()).unwrap();
+        user_addresses.push(cosmos_address);
+        amounts.push(amount.into())
+    }
+    (user_addresses, amounts)
+}
+
+fn generate_valid_accounts_and_amounts(
+    rng: &mut ThreadRng,
+    max: Uint256,
+) -> (Vec<CosmosAddress>, Vec<u64>) {
+    let (user_addresses, mut amounts) = generate_accounts_and_amounts(rng);
+    while total_array_u256(&amounts) > max {
```

Comment:
I hope this doesn't take too long. Very creative!

Reply:
it only takes about 10 seconds on my machine, we'll see how CI likes it.

Reply:
It only took 15 minutes to run, so either not bad or very lucky.